src/Utilities/TimeManager.cs

```diff
@@ -1,11 +1,63 @@
 ﻿using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
 
 namespace ePiggyWeb.Utilities
 {
     public static class TimeManager
     {
         public static DateTime OneMonthAhead { get; }= GetEndOfTheMonth(DateTime.Today.AddMonths(1));
 
+        public static void GetDate(HttpRequest request, out DateTime pageStartDate, out DateTime pageEndDate)
+        {
+            var startDateCookie = request.Cookies["StartDate"];
+            var endDateCookie = request.Cookies["EndDate"];
+
+            if (startDateCookie is null || endDateCookie is null)
+            {
+                var today = DateTime.Now;
+                pageStartDate = new DateTime(today.Year, today.Month, 1);
+                pageEndDate = DateTime.Today;
+            }
+            else
+            {
+                if (DateTime.TryParse(startDateCookie, out var tempStartDate) &&
+                    DateTime.TryParse(endDateCookie, out var tempEndDateTime))
+                {
+                    pageStartDate = tempStartDate;
+                    pageEndDate = tempEndDateTime;
+                }
+                else
+                {
+                    var today = DateTime.Now;
+                    pageStartDate = new DateTime(today.Year, today.Month, 1);
+                    pageEndDate = DateTime.Today;
+                }
+            }
+        }
+
+        public static void SetDate(DateTime startDate, DateTime endDate, ref string errorMessage, HttpResponse response, HttpRequest request, out DateTime pageStartDate, out DateTime pageEndDate)
+        {
+            if (startDate > endDate)
+            {
+                errorMessage = "Start date is bigger than end date!";
+                GetDate(request, out pageStartDate, out pageEndDate);
+            }
+            else
+            {
+                pageStartDate = startDate;
+                pageEndDate = endDate;
+                response.Cookies.Delete("StartDate");
+                response.Cookies.Delete("EndDate");
+                var cookieOptions = new CookieOptions
+                {
+                    Expires = DateTimeOffset.UtcNow.AddMinutes(15)
```

Comment:
Not sure about the option, since this allows filters to be saved only for 15 mins. Was this intended?

Reply:
I really don't know how long these filtering options should last, so now they just last enough for active testing. I'm open to suggestions on how long they should stay :)

Reply:
I guess until logging off? Since they can be changed and would be nice if stayed consistent through whole "workflow of the user". And then reset until next time.

Reply:
I'm not sure how to delete the cookies on the automatic sign out ( when the authentication cookies expire), but I can make the filter cookies last a long time and have them removed when manually signing out

Reply:
I tried to make these cookies sliding just like the authorization cookies, however that proved quite difficult to implement and I don't think it's worth the struggle :D

Reply:
I think then the most logical option is like you've said give them lifetime of aroun 3-5 hours and leave them as it is.

Reply:
Changed them to stay on for 4 hours